distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Fan_Out.enso

```diff
@@ -118,11 +134,35 @@ fan_out_to_rows_and_columns table input_column_id function column_names at_least
 
     0.up_to num_input_rows . each i->
         input_value = input_storage.getItemBoxed i
-        output_values = function input_value |> maybe_add_empty_row
-        # Append each group of values to the builder.
-        output_values.each row_unchecked->
-            row = uniform_length num_output_columns row_unchecked problem_builder
-            row.each_with_index i-> v-> output_column_builders.at i . append v
+        output_values = maybe_add_empty_row (function input_value)
+
+        case dynamic_columns of
```

Comment:
The complexity of `fan_out_to_rows_and_columns` has gotten really high and IMHO it has reached a tipping point after which it is quite hard to understand what is going on here - I need to read the code back-and-forth to understand the relations between all these mappings etc.

This method contains 2 code paths: for dynamic and static set of columns that are intertwined and one has to carefully read each variable initialization to understand the purpose it serves in each mode.

The logic for the dynamic and static approach has very little in common really.

I think we should split this into 2 separate functions - one for dynamic and one for the static case. Then it reduce the overall complexity and make it easier to read. We can factor out the common pieces as another helper functions, making the 'units' even smaller and easier to grasp.

Reply:
don't disagree was just aiming for smallest change here but will break in two.